### PR TITLE
Classify MNIST handwritten digits with Keras

### PR DESCRIPTION
**Pull request by keogh**
* Finish post

* Add images for the post

* Add configuration to the config.yml to allow post assets, this will
create a folder in the `_posts` folder with the same name as the post
when using the command `> hexo new post-title` https://hexo.io/docs/asset-folders.html#Post-Asset-Folder

### CODE REVIEW

File: source/_posts/classify-mnist-digits-keras.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+3. Visualize data to get some intuition.
+4. Prepare input data to feed the input layer.
+5. Prepare label data.
+
+```python
+# load dataset
+(X_train, y_train), (X_test, y_test) = mnist.load_data();
+```
+
+The `load_data()` function return two tuples, already divinding the data in training and test collections.
```
Comment (hyeomans) [inline source/_posts/classify-mnist-digits-keras.md]: Is it divinding? or dividing?

File: source/_posts/classify-mnist-digits-keras.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+The `X_train.shape` will be `(60000, 28, 28)`, 60,000 examples of 28x28 pixes grayscale images of the 10 digits. While `y_train` is an vector fo 60,000 examples where each element is an integer from 0 to 9 (`X_test` and `y_test` contains 10,000 examples). Remember that `X_train[n]` corresponds to `y_train[n]`.
```
Comment (hyeomans) [inline source/_posts/classify-mnist-digits-keras.md]: One tip on parentheses, nothing to change: http://thecopyfox.com/2012/11/writing-tip-do-you-overuse-parentheses/

File: source/_posts/classify-mnist-digits-keras.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+plt.imshow(X_train[1], cmap=plt.get_cmap('gray'))
+plt.subplot(223)
+plt.imshow(X_train[2], cmap=plt.get_cmap('gray'))
+plt.subplot(224)
+plt.imshow(X_train[3], cmap=plt.get_cmap('gray'))
+```
+
+<img src="/classify-mnist-digits-keras/pyplot-digits.png" width="300px" />
+
+The next step is to reshape the input matrix to have th shape `batch x height x width x channels`, since these are grayscale images the *channles* will be 1, in other images the *channels* normally would be 3, one for each color in the RGB color model.
```
Comment (hyeomans) [inline source/_posts/classify-mnist-digits-keras.md]: ...to have the shape...

Comment (hyeomans) [reply source/_posts/classify-mnist-digits-keras.md]: `since these are grayscale images the *channles*` -> channels

File: source/_posts/classify-mnist-digits-keras.md [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+After that I connected another `Conv2D` with ReLU activation, 15 filters and 3x3 window and connected it to a 2x2 `MaxPooling2D` layer.
+
+I use a [Dropout](https://keras.io/layers/core/#dropout) layer for regularization to prevent overfitting.
+
+After the convolutions I want to classify the data with fully connected layers, thus I use a [Flatten](https://keras.io/layers/core/#flatten) layer to flat the `Dropout` output to a 1-dimensional vector to be used as input for two [Dense](https://keras.io/layers/core/#dense) (fully-connected) layers with ReLU activation and with 128 and 50 nodes respectively.
+
+The output layer is a `Dense` layer with 10 nodes (same dimension as an element in `Y_train` and `Y_test`) and [softmax](https://en.wikipedia.org/wiki/Softmax_function) activation function.
+
+I could use more or less Conv2D layers, or add other Dropout between the MaxPooling2D and the second Conv2D, I could change the hyperparameters in each layer, or the total nodes in one of the Dense layers, or use one Dense layer instead of two. Trying several architechtures and mesuring which generalize better is part of the Data Scientiest/Machine Learning Engineer daily job.
```
Comment (hyeomans) [inline source/_posts/classify-mnist-digits-keras.md]: Data Scientiest -> Data Scientist

**Review by hyeomans — CHANGES_REQUESTED** [review]

File: source/_posts/classify-mnist-digits-keras.md (hunk 1 above)
Comment (keogh) [reply source/_posts/classify-mnist-digits-keras.md]: alv, dividing

Comment (keogh) [reply source/_posts/classify-mnist-digits-keras.md]: I'm gonna use splitting instead

**Comment by keogh** [comment]
Changes addressed

**Review by hyeomans — APPROVED** [review]
